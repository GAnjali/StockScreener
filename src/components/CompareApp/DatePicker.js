import React, {Component} from "react";
import {SingleDatePicker} from "react-dates";
import 'react-dates/initialize';
import 'react-dates/lib/css/_datepicker.css';
import "../../styles/react_dates_overrides.css"
import "../../styles/styles.css";

class DatePicker extends Component {
    state = {
        focusedInput: false,
    };

    handleFocusChange = () => this.setState({focusedInput: true});

    handleDateChange = (date) => {
        this.setState({
            focusedInput: false
        }, this.props.handleDateChange(date));
    };

    render() {
        return (
            <div>
                <h3 className={"date-picker-label"}>{"Select Date:"}</h3>
                <div className={"single-date-picker"}>
                    <SingleDatePicker
                        date={this.props.date}
                        onDateChange={this.handleDateChange}
                        focused={this.state.focusedInput}
                        onFocusChange={this.handleFocusChange}
                        numberOfMonths={1}
                    />
                </div>
            </div>
        )
    }
}

export default DatePicker;