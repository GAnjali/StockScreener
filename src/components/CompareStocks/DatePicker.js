import React, {Component} from "react";
import {SingleDatePicker} from "react-dates";
import 'react-dates/initialize';
import 'react-dates/lib/css/_datepicker.css';
import "../../styles/react_dates_overrides.css"
import "../../styles/styles.css";

class DatePicker extends Component {
    state = {
        focusedInput: false,
    };

    handleFocusChange = ({focused}) => this.setState({focusedInput: focused});

    handleDateChange = (date) => {
        this.props.handleDateChange(date);
    };

    render() {
        return (
            <div>
                <h3 className={"date-picker-label"}>{"Select Date:"}</h3>
                <div className={"single-date-picker"}>
                    <SingleDatePicker
                        id='date'
                        date={this.props.date}
                        onDateChange={this.handleDateChange}
                        focused={this.state.focusedInput}
                        onFocusChange={this.handleFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                </div>
            </div>
        )
    }
}

export default DatePicker;