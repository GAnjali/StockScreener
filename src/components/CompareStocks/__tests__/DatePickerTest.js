import {shallow} from "enzyme";
import React from "react";
import DatePicker from "../DatePicker";

describe("DatePickerTests", () => {
    it("Should render without crashing", () => {
        shallow(<DatePicker/>);
    });

    it('should match the DatePicker snapshot', function () {
        const dashboardComponent = shallow(<DatePicker/>);
        expect(dashboardComponent).toMatchSnapshot();
    });
});