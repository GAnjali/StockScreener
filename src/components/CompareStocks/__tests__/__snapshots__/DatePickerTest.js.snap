// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerTests should match the DatePicker snapshot 1`] = `ShallowWrapper {}`;

exports[`DatePickerTests should match the InputField snapshot 1`] = `ShallowWrapper {}`;
